tests/Microsoft.DotNet.Framework.Docker.Tests/ImageTests.cs

```diff
@@ -260,6 +281,33 @@ private string GetImage(string imageType, string version, string osVariant)
             return image;
         }
 
+        private static string GetRegistryName(string repo, string tag)
+        {
+            bool imageExistsInStaging = true;
+
+            // In the case of running this in a local development environment, there would likely be no image info file
+            // provided. In that case, the assumption is that the images exist in the staging location.
+
+            if (ImageTests.ImageInfoData.Value != null)
+            {
+                JObject repoInfo = (JObject)ImageTests.ImageInfoData.Value
+                    .FirstOrDefault(imageInfoRepo => imageInfoRepo["repo"].ToString() == repo);
+
+                if (repoInfo["images"] != null)
+                {
+                    imageExistsInStaging = repoInfo["images"]
+                        .Cast<JProperty>()
+                        .Any(imageInfo => imageInfo.Value["simpleTags"].Any(imageTag => imageTag.ToString() == tag));
+                }
+                else
+                {
+                    imageExistsInStaging = false;
+                }
+            }
+
+            return imageExistsInStaging ? $"{Registry}/{RepoPrefix}" : "mcr.microsoft.com/";
```

Comment:
Would it make sense to assert/validate that `Registry` and `RepoPrefix` have been set in the staging code path?